apps/speed.c

```diff
@@ -1778,7 +1778,12 @@ int speed_main(int argc, char **argv)
         buflen = lengths[size_num - 1];
         if (buflen < 36)    /* size of random vector in RSA benchmark */
             buflen = 36;
-        buflen += MAX_MISALIGNMENT + 1;
+        if (0x7fffffff - (MAX_MISALIGNMENT + 1) < buflen)
```

Comment:
This should be `INT_MAX` not `0x7fffffff`.

Reply:
What about using `buflen + MAX_MISALIGNMENT + 1 < buflen`?

Reply:
@paulidale fixed. Also changed in a couple of other places where `0x7fffffff` is used.

Reply:
@t8m C doesn't actually mandate signed integer overflow to be wraparound. I'm pretty sure the platforms OpenSSL supports will indeed wraparound, but I did this just to be safe.

Reply:
C specifies that signed integer overflow is undefined.  We don't want to rely on [undefined behaviour](https://t.co/0oVHLskwKN).

Reply:
Not only does C not mandate signed integer overflow to be wraparound, signed integer overflow is explicitly considered undefined behavior and is to be avoided at all costs.